src/test/java/org/prebid/server/bidder/trustedstack/TrustedstackBidderTest.java

```diff
@@ -0,0 +1,246 @@
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
+                .containsExactly(bidRequest);
+    }
+
+    @Test
+    public void makeBidsShouldReturnErrorIfResponseBodyCouldNotBeParsed() {
+        // given
+        final BidderCall<BidRequest> httpCall = sampleHttpCall(givenBidRequest(), "invalid response");
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1)
+                .allMatch(error -> error.getType() == BidderError.Type.bad_server_response
+                        && error.getMessage().startsWith("Failed to decode: Unrecognized token"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyListIfBidResponseIsNull() throws JsonProcessingException {
+        // given
+        final BidderCall<BidRequest> httpCall;
+        httpCall = sampleHttpCall(givenBidRequest(), mapper.writeValueAsString(null));
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyListIfBidResponseSeatBidIsNull() throws JsonProcessingException {
+        // given
+        final BidderCall<BidRequest> httpCall;
+        httpCall = sampleHttpCall(null, mapper.writeValueAsString(BidResponse.builder().build()));
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnBannerBidIfBannerIsPresent() throws JsonProcessingException {
+        // given
+        final BidderCall<BidRequest> httpCall = sampleHttpCall(
+                givenBidRequest(),
+                mapper.writeValueAsString(sampleBidResponse(bidBuilder -> bidBuilder.impid("123"))));
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .containsExactly(BidderBid.of(Bid.builder().impid("123").build(), banner, "USD"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnCorrespondingMtypesAndAdTypes() throws JsonProcessingException {
+        final List<Bid> bids = new ArrayList<>();
+        final Bid bid1 = Bid.builder().impid("imp_id").mtype(1).build();
+        final Bid bid2 = Bid.builder().impid("imp_id").mtype(2).build();
+        final Bid bid3 = Bid.builder().impid("imp_id").mtype(3).build();
+        final Bid bid4 = Bid.builder().impid("imp_id").mtype(4).build();
+        bids.add(bid1);
```

Comment:
We would prefer to have separate test for each mtype

Reply:
Done.

Reply:
@SerhiiNahornyi requesting review as the requested changes has been implemented.